src/cpp/fastdds/builtin/type_lookup_service/TypeLookupRequestListener.cpp

```diff
@@ -40,104 +36,406 @@ namespace fastdds {
+        EPROSIMA_LOG_WARNING(TYPELOOKUP_SERVICE_REQUEST_LISTENER, "Error getting type dependencies.");
+        answer_request(request_id, rpc::RemoteExceptionCode_t::REMOTE_EX_INVALID_ARGUMENT);
+    }
+}
 
-                typelookup_manager_->send_reply(*reply);
-                typelookup_manager_->reply_type_.deleteData(reply);
+TypeLookup_getTypeDependencies_Out TypeLookupRequestListener::prepare_get_type_dependencies_response(
+        const xtypes::TypeIdentifierSeq& id_seq,
+        const std::unordered_set<xtypes::TypeIdentfierWithSize>& type_dependencies,
+        const std::vector<uint8_t>& continuation_point)
+{
+    TypeLookup_getTypeDependencies_Out out;
 
-                break;
+    // Check if all dependencies can be sent in a single response
+    if (type_dependencies.size() < MAX_DEPENDENCIES_PER_REPLY)
+    {
+        for (const auto& type_identifier : type_dependencies)
+        {
+            out.dependent_typeids().emplace_back(type_identifier);
+        }
+    }
+    else
+    {
+        size_t start_index = 0;
+        // Check if a continuation point is provided, and calculate starting point if there is
+        if (!continuation_point.empty())
+        {
+            start_index = calculate_continuation_point(continuation_point) * MAX_DEPENDENCIES_PER_REPLY;
+        }
+
+        // Copy the dependencies within the specified range directly to out
+        auto start_it = std::next(type_dependencies.begin(), start_index);
+        auto end_it = std::next(start_it, std::min<size_t>(MAX_DEPENDENCIES_PER_REPLY,
+                        type_dependencies.size() - start_index));
+        for (auto it = start_it; it != end_it; ++it)
+        {
+            out.dependent_typeids().emplace_back(*it);
+        }
+
+        if ((start_index + MAX_DEPENDENCIES_PER_REPLY) > type_dependencies.size())
+        {
+            // If all dependent types have been sent, remove from map
+            std::lock_guard<std::mutex> lock(requests_with_continuation_mutex_);
+            auto requests_it = requests_with_continuation_.find(id_seq);
+            if (requests_it != requests_with_continuation_.end())
+            {
+                requests_with_continuation_.erase(requests_it);
             }
-            default:
-                break;
+        }
+        else
+        {
+            // Set the continuation point for the next request
+            out.continuation_point(create_continuation_point(calculate_continuation_point(continuation_point) + 1));
         }
     }
+
+    return out;
+}
+
+void TypeLookupRequestListener::answer_request(
+        SampleIdentity request_id,
+        rpc::RemoteExceptionCode_t exception_code,
+        TypeLookup_getTypeDependencies_Out& out)
+{
+    TypeLookup_Reply* reply = static_cast<TypeLookup_Reply*>(typelookup_manager_->reply_type_.createData());
+
+    TypeLookup_getTypeDependencies_Result result;
+    result.result(out);
+    reply->return_value().getTypeDependencies(result);
```

Comment:
Why use an auxiliary variable?

```suggestion
    TypeLookup_getTypeDependencies_Result result;
    result.result(out);
    reply->return_value().getTypeDependencies().result(out);
```

Please, apply to other similar implementations.

Reply:
it necessary to the type of TypeLookup_Return